plugins/vbe/fu-vbe-simple-device.c

```diff
@@ -0,0 +1,844 @@
+/*
+ * VBE plugin for fwupd,vbe-simple
+ *
+ * Copyright (C) 2017 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2022 Google LLC
+ * Written by Simon Glass <sjg@chromium.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <ctype.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <libfdt.h>
+#include <linux/fs.h>
+#include <stdio.h>
+#include <sys/ioctl.h>
+
+#include "fit.h"
+#include "fu-dfu-common.h"
+#include "fu-vbe-simple-device.h"
+
+/* G_OBJECT properties associated with the plugin */
+enum { PROP_0, PROP_DEVNAME, PROP_AREA_START, PROP_AREA_SIZE, PROP_LAST };
+
+/**
+ * struct vbe_simple_state - current state of this VBE method
+ *
+ * This simply records successful updates at present. There is no log of
+ * failures.
+ *
+ * @finish_time: Time that the the last update happened, 0 if none
+ * @cur_version: Currently installed version, NULL if none
+ * @status: State of the last update (always "completed"), NULL if none
+ */
+struct vbe_simple_state {
+	/* information about the last update */
+	time_t finish_time;
```

Comment:
@sjg20 I'm a bit confused why we're setting `finish_time`; is it going to be used in the future?

Reply:
It allows the firmware to show when the last update was done. We can drop it for now if you like